src/sql/src/error.rs

```diff
@@ -285,7 +301,9 @@ impl ErrorExt for Error {
             | ConvertToLogicalExpression { .. }
             | Simplification { .. }
             | InvalidInterval { .. }
-            | PermissionDenied { .. } => StatusCode::InvalidArguments,
+            | PermissionDenied { .. }
```

Comment:
Not related to this PR, but `PermissionDenied` should return `StatusCode:: PermissionDenied`.